techniques/systemSettings/remoteAccess/sshKeyDistribution/3.0/sshKeyDistribution.st

```diff
@@ -60,6 +60,20 @@ bundle agent check_ssh_key_distribution
         string => execresult("${paths.echo} '${sshkey_distribution_key[${sshkey_distribution_index}]}' | ${paths.sed} -E 's/(.*\s+)?(${ssh_types})\s+(\S+)(\s.*)?/\3/'", "useshell"),
         ifvarclass => "correct_ssh_key_format_${sshkey_distribution_index}";
 
+
+      # Default home dir mode
+      "homedir_mode_to_use" string => "700";
+
+    etc_login_defs_exists::
+      # read the /etc/login.defs to get umask for home creation if home dir is not there
+      "nb_line_read" int => readstringarray("login_defs_content", "/etc/login.defs","#[^\n]*", "\s+", "99999", "9999999");
+      # Use the umask to get the value
+      "homedir_mode_real" string => eval("777-${login_defs_content[UMASK][1]}}", "math", "infix");
+      # Format to int
+      "homedir_mode_int" string => format("%0.0f", ${homedir_mode_real});
+      "homedir_mode_to_use" string => "${homedir_mode_int}";
```

Comment:
Why not doing this in one line?
"homedir_mode_to_use" string => format("%0.0f", ${homedir_mode_real});

Reply:
because homedir_mode_real may not exist, in this case, the format is not evaluated, and i use the default home_dir_to_use (line 65)
i'm not sure that doing it in one step may not cancel the actual value